src/ir/context.rs

```diff
@@ -1038,13 +1105,14 @@ impl<'ctx> BindgenContext<'ctx> {
                         let sub_item = Item::new(sub_id,
                                                  None,
                                                  None,
-                                                 template_decl_id,
+                                                 self.current_module,
```

Comment:
So we'll instantiate templates against the current module? Even if they're nested? That looks suspicious. The parentage relationship for...

```rust
struct Foo {
    template<typename T> struct Bar { T baz; };
    Bar<int> m_member;
};
```

So, if I'm reading this correctly, we'll scope `Bar<int>` to the gloal scope, not to `Foo`, right? Can you justify that? If it doesn't matter (which I think it could be true). Could you add a comment?